ksql-serde/src/main/java/io/confluent/ksql/serde/Format.java

```diff
@@ -117,25 +106,18 @@ public void validateProperties(final Map<String, String> properties) {
    * created with that stream as its source will also have the same delimiter
    * value of {@code x}</p>
    *
+   * <p>The default implementation of this method assumes that all of the
+   * supported properties are inheritable.</p>
+   *
    * @return the set of properties that are considered "inheritable"
    */
-  public Set<String> getInheritableProperties() {
-    return inheritableProperties;
+  default Set<String> getInheritableProperties() {
+    return getSupportedProperties();
   }
 
   /**
    * @param info the info containing information required for generating the factory
    * @return a {@code KsqlSerdeFactory} that generates serdes for the given format
    */
-  public abstract KsqlSerdeFactory getSerdeFactory(FormatInfo info);
-
-  public static Format of(final FormatInfo value) {
-    try {
-      final Format format = valueOf(value.getFormat().toUpperCase());
-      format.validateProperties(value.getProperties());
-      return format;
-    } catch (final IllegalArgumentException e) {
-      throw new KsqlException("Unknown format: " + value.getFormat());
-    }
-  }
+  KsqlSerdeFactory getSerdeFactory(FormatInfo info);
```

Comment:
If `KsqlSerdeFactory` is going to become part of our public API then we should give it some love to make it more extensible, e.g.:

 * Rather than passing config and SR clients as params, pass them via a Context object, which allows us to add more things to it later without needing to change the signature. 
 * Consider removing KSqlConfig from the method sigs completely, as this is specific to our serde classes. Instead, pluggable formats can implement `Configurable` if they need config.  (We should limit _what_ config is passed to 3rd party formats just like we do for UDFs e.g. we shouldn't be passing all config that may contain secrets etc).
* move the type checking version of `createSerde` out of the interface.

Reply:
https://github.com/confluentinc/ksql/issues/4441